### PR TITLE
Tomcox/resourcepart

### PR DESCRIPTION
**Pull request by tomacox**
documentation for ResourcePart

### CODE REVIEW

**Comment by mikekinsman** [comment]
Docs Build status updates of commit _[99e6638](https://github.com/Azure/portaldocs/pull/236/commits/99e66382656f19c6f1445b0ea35f26d0d4556b14)_: 

### :x: Validation status: errors

Please follow instructions [here](https://review.docs.microsoft.com/en-us/help/onboard/faq?branch=master#build--publish) which may help to resolve issue.


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[]() | :x:Error | | [Details](#user-content-3b4c23a1)

<a id="3b4c23a1"></a>
###
 - **[Error]** `````Publish service's config file: .openpublishing.publish.config.json is not found. Exception: `````

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2019%5C6%5C7%5C3d86255f-f488-188e-02a3-9e92aa67830d%5CPullRequest%5C201906070612029283-236%5Cworkflow_report.html?sv=2016-05-31&sr=b&sig=Ys%2FltlrOvSxW%2FBTkMSguSaB0DfS0vguJoOAbrZZQrn4%3D&st=2019-06-07T06%3A07%3A18Z&se=2019-07-08T06%3A12%3A18Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

File: portal-sdk/templates/top-extensions-parts-resourcepart.md [hunk 1]
```diff
@@ -0,0 +1,104 @@
+## ResourcePart
```
Comment (sewatson) [inline portal-sdk/templates/top-extensions-parts-resourcepart.md]: Headings should have a new line after them before content

File: portal-sdk/templates/top-extensions-parts-resourcepart.md [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+  <AssetType Name="Author"
+             ViewModel="{ViewModel Name=AuthorViewModel, Module=./Document/AssetViewModels/AuthorViewModel}"
+             CompositeDisplayName="{Resource AssetTypeNames.Author, Module=ClientResources}"
+             Icon="{Svg IsLogo=true, File=../../../Svg/Documents/author.svg}"
+             BladeName="AuthorBlade"
+             PartName="{ResourcePart}">
+```
+
+### Pinning a blade
+Blades which represent a resource can also be pinned with the resource part.  This is supported for both noPDL blades and legacy blades.  The blades only required parameter must be named 'id' and a string type.  To pin a modern noPDL blade simply return a part reference for the ResourcePart from the onPin method as in this example:
```
Comment (sewatson) [inline portal-sdk/templates/top-extensions-parts-resourcepart.md]: Maybe include a link to the pinning documentation?

File: portal-sdk/templates/top-extensions-parts-resourcepart.md [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+
+### Redirecting existing parts on dashboards
+If this is a existing asset type for which the ResourcePart is replacing a custom part then a redirect will also need to be added.   The redirect will change these custom part instances on dashboards to be the new ResourcePart type.  The redirect is declared in PDL.   It can be located in any PDL file though for organizational purposes it would best to place it in the same file which the AssetType is declared.   The PDL to redirect existing pinned parts to the new ResourcePart looks like this:
+
+```xml
+  <RedirectPart Name="AuthorPart">
+    <ResourcePart />
+  </RedirectPart>
+```
+## Displaying resource status
```
Comment (sewatson) [inline portal-sdk/templates/top-extensions-parts-resourcepart.md]: This should be level 3 heading `###`

Comment (sewatson) [reply portal-sdk/templates/top-extensions-parts-resourcepart.md]: Make sure the sub categories below are an extra level if needed.

**Comment by mikekinsman** [comment]
Docs Build status updates of commit _[42571a7](https://github.com/Azure/portaldocs/pull/236/commits/42571a7a5d755a4435638bc82e20076056167591)_: 

### :x: Validation status: errors

Please follow instructions [here](https://review.docs.microsoft.com/en-us/help/onboard/faq?branch=master#build--publish) which may help to resolve issue.


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[]() | :x:Error | | [Details](#user-content-43d20dc5)

<a id="43d20dc5"></a>
###
 - **[Error]** `````Publish service's config file: .openpublishing.publish.config.json is not found. Exception: `````

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2019%5C6%5C7%5C3d86255f-f488-188e-02a3-9e92aa67830d%5CPullRequest%5C201906072003116161-236%5Cworkflow_report.html?sv=2016-05-31&sr=b&sig=QxBHdYw7VjvGmrF%2BEJo7w3uwnlpgoMB9uidlg2Nb1e8%3D&st=2019-06-07T19%3A58%3A24Z&se=2019-07-08T20%3A03%3A24Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Comment by mikekinsman** [comment]
Docs Build status updates of commit _[77e0dcb](https://github.com/Azure/portaldocs/pull/236/commits/77e0dcb5a1980088b4bed4d030c186c7da58bea5)_: 

### :x: Validation status: errors

Please follow instructions [here](https://review.docs.microsoft.com/en-us/help/onboard/faq?branch=master#build--publish) which may help to resolve issue.


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[]() | :x:Error | | [Details](#user-content-f1ee9501)

<a id="f1ee9501"></a>
###
 - **[Error]** `````Publish service's config file: .openpublishing.publish.config.json is not found. Exception: `````

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2019%5C6%5C9%5C3d86255f-f488-188e-02a3-9e92aa67830d%5CPullRequest%5C201906092327548123-236%5Cworkflow_report.html?sv=2016-05-31&sr=b&sig=hLIMmNYR0wxqpaOZ%2FyK%2BmEa5uFO2PO3JglcxHe6LncU%3D&st=2019-06-09T23%3A23%3A08Z&se=2019-07-10T23%3A28%3A08Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Pull request merged**